tests/UI/FactoryAbstractTest.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+require_once("libs/composer/vendor/autoload.php");
+
+/**
+ *	Defines tests every UI-factory should pass.
+ *	
+ */
+abstract class FactoryAbstractTest extends PHPUnit_Framework_TestCase {
+    abstract public function getFactoryInstance();
+
+    /**
+     *	This will be used in tests of the UI component creation of a factory.
+     *	Make sure all the UI Components, which your factory may create,
+     *	are listed here by the format array("methodName" => mixed[] $args).
+     *	If a method takes no parameters, use "methodName" => array().
+     *	If you skip any methods, assertions will fail...
+     *
+     *	@return	mixed["methodName"][]
+	 */
+    abstract protected function UIComponentCreationArguments();
+
+	public function test_any_interfaces_are_factories() {
+		if(count($this->factoryInterfaceReflections($this->getFactoryInstance())) > 0) {
+			$this->assertTrue(true);
+		} else {
+			$this->assertFalse('no factory interfaces found');
+		}
+	}
+
+	public function test_ui_component_creation() {
+		$f = $this->getFactoryInstance();
+		$ui_component_creation_arguments = $this->UIComponentCreationArguments();
+
+		$method_reflections = array();
+
+		foreach ($this->getFactoryInterfaceReflections($f) as $interface_name => $interface_reflection) {
+			$method_reflections = array_merge($method_reflections, $interface_reflection->getMethods());
+		}
+
+		foreach ($method_reflections as $method_reflection) {
+			if(isset($ui_component_creation_arguments[$method_reflection->getName()])) {
+				$possible_ui_component = $method_reflection->invokeArgs($f,$ui_component_creation_arguments[$method_reflection->getName()]);
+				$this->assertTrue($possible_ui_component instanceof \ILIAS\UI\Cmponent);
```

Comment:
assertInstanceOf

Bitte mal in die PHPUnit-Dokumentation reinschauen, damit du einen Überblick über die verschiedenen assertXY-Methoden bekommst.